### ISSUE TITLE
Add more dependencies to the copy-builtin.sh script to avoid kernel build failures 

### ISSUE THREAD

**Issue by morfikov** [openwall/lkrg#34]
I wanted to try the in-kernel build script, but I got the following error (I just changed the KDIR var):

```
$ ./scripts/copy-builtin.sh
Copying /media/debuilder/git-lkrg/lkrg/scripts/../src/* to /media/debuilder/git-kernel/linux-5.10/security/lkrg along with Kconfig:
# SPDX-License-Identifier: GPL-2.0-only
config SECURITY_LKRG
        tristate "LKRG support"
        depends on SECURITY
        default m
        help
          This selects LKRG - Linux Kernel Runtime Guard, which provides
          integrity validation and anti-exploitation functions.

          If you are unsure how to answer this question, answer M.

and Makefile
# SPDX-License-Identifier: GPL-2.0-only

obj-$(CONFIG_SECURITY_LKRG) := p_lkrg.o
p_lkrg-objs += modules/ksyms/p_resolve_ksym.o \
                  modules/hashing/p_lkrg_fast_hash.o \
                  modules/comm_channel/p_comm_channel.o \
                  modules/integrity_timer/p_integrity_timer.o \
                  modules/kmod/p_kmod.o \
                  modules/database/CPU.o \
                  modules/database/arch/x86/p_x86_metadata.o \
                  modules/database/arch/x86/p_switch_idt/p_switch_idt.o \
                  modules/database/arch/arm64/p_arm64_metadata.o \
                  modules/database/arch/arm/p_arm_metadata.o \
                  modules/database/arch/p_arch_metadata.o \
                  modules/database/JUMP_LABEL/p_arch_jump_label_transform/p_arch_jump_label_transform.o \
                  modules/database/JUMP_LABEL/p_arch_jump_label_transform_apply/p_arch_jump_label_transform_apply.o \
                  modules/database/FTRACE/p_ftrace_modify_all_code/p_ftrace_modify_all_code.o \
                  modules/database/FTRACE/p_ftrace_enable_sysctl/p_ftrace_enable_sysctl.o \
                  modules/database/p_database.o \
                  modules/notifiers/p_notifiers.o \
                  modules/self-defense/hiding/p_hiding.o \
                  modules/exploit_detection/p_rb_ed_trees/p_rb_ed_pids/p_rb_ed_pids_tree.o \
                  modules/exploit_detection/syscalls/p_install.o \
                  modules/exploit_detection/syscalls/p_search_binary_handler/p_search_binary_handler.o \
                  modules/exploit_detection/syscalls/p_call_usermodehelper/p_call_usermodehelper.o \
                  modules/exploit_detection/syscalls/p_call_usermodehelper_exec/p_call_usermodehelper_exec.o \
                  modules/exploit_detection/syscalls/p_do_exit/p_do_exit.o \
                  modules/exploit_detection/syscalls/p_wake_up_new_task/p_wake_up_new_task.o \
                  modules/exploit_detection/syscalls/p_sys_setuid/p_sys_setuid.o \
                  modules/exploit_detection/syscalls/p_sys_setreuid/p_sys_setreuid.o \
                  modules/exploit_detection/syscalls/p_sys_setresuid/p_sys_setresuid.o \
                  modules/exploit_detection/syscalls/p_sys_setfsuid/p_sys_setfsuid.o \
                  modules/exploit_detection/syscalls/p_sys_setgid/p_sys_setgid.o \
                  modules/exploit_detection/syscalls/p_sys_setregid/p_sys_setregid.o \
                  modules/exploit_detection/syscalls/p_sys_setresgid/p_sys_setresgid.o \
                  modules/exploit_detection/syscalls/p_sys_setfsgid/p_sys_setfsgid.o \
                  modules/exploit_detection/syscalls/p_set_current_groups/p_set_current_groups.o \
                  modules/exploit_detection/syscalls/p_generic_permission/p_generic_permission.o \
                  modules/exploit_detection/syscalls/p_sel_write_enforce/p_sel_write_enforce.o \
                  modules/exploit_detection/syscalls/p_seccomp/p_seccomp.o \
                  modules/exploit_detection/syscalls/p_sys_unshare/p_sys_unshare.o \
                  modules/exploit_detection/syscalls/p_sys_setns/p_sys_setns.o \
                  modules/exploit_detection/syscalls/caps/p_sys_capset/p_sys_capset.o \
                  modules/exploit_detection/syscalls/caps/p_cap_task_prctl/p_cap_task_prctl.o \
                  modules/exploit_detection/syscalls/keyring/p_key_change_session_keyring/p_key_change_session_keyring.o \
                  modules/exploit_detection/syscalls/keyring/p_sys_add_key/p_sys_add_key.o \
                  modules/exploit_detection/syscalls/keyring/p_sys_request_key/p_sys_request_key.o \
                  modules/exploit_detection/syscalls/keyring/p_sys_keyctl/p_sys_keyctl.o \
                  modules/exploit_detection/syscalls/p_security_ptrace_access/p_security_ptrace_access.o \
                  modules/exploit_detection/syscalls/compat/p_compat_sys_keyctl/p_compat_sys_keyctl.o \
                  modules/exploit_detection/syscalls/compat/p_compat_sys_capset/p_compat_sys_capset.o \
                  modules/exploit_detection/syscalls/compat/p_compat_sys_add_key/p_compat_sys_add_key.o \
                  modules/exploit_detection/syscalls/compat/p_compat_sys_request_key/p_compat_sys_request_key.o \
                  modules/exploit_detection/syscalls/__x32/p_x32_sys_keyctl/p_x32_sys_keyctl.o \
                  modules/exploit_detection/syscalls/override/p_override_creds/p_override_creds.o \
                  modules/exploit_detection/syscalls/override/p_revert_creds/p_revert_creds.o \
                  modules/exploit_detection/syscalls/override/overlayfs/p_ovl_create_or_link/p_ovl_create_or_link.o \
                  modules/exploit_detection/syscalls/pCFI/p_mark_inode_dirty/p_mark_inode_dirty.o \
                  modules/exploit_detection/syscalls/pCFI/p_schedule/p_schedule.o \
                  modules/exploit_detection/syscalls/pCFI/p___queue_work/p___queue_work.o \
                  modules/exploit_detection/syscalls/pCFI/p_lookup_fast/p_lookup_fast.o \
                  modules/exploit_detection/syscalls/p_ttwu_do_wakeup/p_ttwu_do_wakeup.o \
                  modules/exploit_detection/syscalls/p_capable/p_capable.o \
                  modules/exploit_detection/syscalls/p_scm_send/p_scm_send.o \
                  modules/exploit_detection/p_exploit_detection.o \
                  p_lkrg_main.o
Commit msg: LKRG in-tree @ 1f9809db4bed4e33fcc3db31
Ctrl+c to quit, any other key to continue

./scripts/copy-builtin.sh: 54: pushd: not found
```
So apparently, pushd is missing, but I couldn't find it in debian sid/exp (I was using apt-file to search for it).

**Comment by morfikov** [comment]
I've manage to solve this by sourcing the script via:

```
$ source scripts/copy-builtin.sh
Copying /media/debuilder/git-lkrg/lkrg/scripts/../src/* to /media/debuilder/git-kernel/linux-5.10/security/lkrg along with Kconfig:
# SPDX-License-Identifier: GPL-2.0-only
config SECURITY_LKRG
        tristate "LKRG support"
        depends on SECURITY
        default m
        help
          This selects LKRG - Linux Kernel Runtime Guard, which provides
          integrity validation and anti-exploitation functions.

          If you are unsure how to answer this question, answer M.

and Makefile
# SPDX-License-Identifier: GPL-2.0-only

obj-$(CONFIG_SECURITY_LKRG) := p_lkrg.o
p_lkrg-objs += modules/ksyms/p_resolve_ksym.o \
                  modules/hashing/p_lkrg_fast_hash.o \
                  modules/comm_channel/p_comm_channel.o \
                  modules/integrity_timer/p_integrity_timer.o \
                  modules/kmod/p_kmod.o \
                  modules/database/CPU.o \
                  modules/database/arch/x86/p_x86_metadata.o \
                  modules/database/arch/x86/p_switch_idt/p_switch_idt.o \
                  modules/database/arch/arm64/p_arm64_metadata.o \
                  modules/database/arch/arm/p_arm_metadata.o \
                  modules/database/arch/p_arch_metadata.o \
                  modules/database/JUMP_LABEL/p_arch_jump_label_transform/p_arch_jump_label_transform.o \
                  modules/database/JUMP_LABEL/p_arch_jump_label_transform_apply/p_arch_jump_label_transform_apply.o \
                  modules/database/FTRACE/p_ftrace_modify_all_code/p_ftrace_modify_all_code.o \
                  modules/database/FTRACE/p_ftrace_enable_sysctl/p_ftrace_enable_sysctl.o \
                  modules/database/p_database.o \
                  modules/notifiers/p_notifiers.o \
                  modules/self-defense/hiding/p_hiding.o \
                  modules/exploit_detection/p_rb_ed_trees/p_rb_ed_pids/p_rb_ed_pids_tree.o \
                  modules/exploit_detection/syscalls/p_install.o \
                  modules/exploit_detection/syscalls/p_search_binary_handler/p_search_binary_handler.o \
                  modules/exploit_detection/syscalls/p_call_usermodehelper/p_call_usermodehelper.o \
                  modules/exploit_detection/syscalls/p_call_usermodehelper_exec/p_call_usermodehelper_exec.o \
                  modules/exploit_detection/syscalls/p_do_exit/p_do_exit.o \
                  modules/exploit_detection/syscalls/p_wake_up_new_task/p_wake_up_new_task.o \
                  modules/exploit_detection/syscalls/p_sys_setuid/p_sys_setuid.o \
                  modules/exploit_detection/syscalls/p_sys_setreuid/p_sys_setreuid.o \
                  modules/exploit_detection/syscalls/p_sys_setresuid/p_sys_setresuid.o \
                  modules/exploit_detection/syscalls/p_sys_setfsuid/p_sys_setfsuid.o \
                  modules/exploit_detection/syscalls/p_sys_setgid/p_sys_setgid.o \
                  modules/exploit_detection/syscalls/p_sys_setregid/p_sys_setregid.o \
                  modules/exploit_detection/syscalls/p_sys_setresgid/p_sys_setresgid.o \
                  modules/exploit_detection/syscalls/p_sys_setfsgid/p_sys_setfsgid.o \
                  modules/exploit_detection/syscalls/p_set_current_groups/p_set_current_groups.o \
                  modules/exploit_detection/syscalls/p_generic_permission/p_generic_permission.o \
                  modules/exploit_detection/syscalls/p_sel_write_enforce/p_sel_write_enforce.o \
                  modules/exploit_detection/syscalls/p_seccomp/p_seccomp.o \
                  modules/exploit_detection/syscalls/p_sys_unshare/p_sys_unshare.o \
                  modules/exploit_detection/syscalls/p_sys_setns/p_sys_setns.o \
                  modules/exploit_detection/syscalls/caps/p_sys_capset/p_sys_capset.o \
                  modules/exploit_detection/syscalls/caps/p_cap_task_prctl/p_cap_task_prctl.o \
                  modules/exploit_detection/syscalls/keyring/p_key_change_session_keyring/p_key_change_session_keyring.o \
                  modules/exploit_detection/syscalls/keyring/p_sys_add_key/p_sys_add_key.o \
                  modules/exploit_detection/syscalls/keyring/p_sys_request_key/p_sys_request_key.o \
                  modules/exploit_detection/syscalls/keyring/p_sys_keyctl/p_sys_keyctl.o \
                  modules/exploit_detection/syscalls/p_security_ptrace_access/p_security_ptrace_access.o \
                  modules/exploit_detection/syscalls/compat/p_compat_sys_keyctl/p_compat_sys_keyctl.o \
                  modules/exploit_detection/syscalls/compat/p_compat_sys_capset/p_compat_sys_capset.o \
                  modules/exploit_detection/syscalls/compat/p_compat_sys_add_key/p_compat_sys_add_key.o \
                  modules/exploit_detection/syscalls/compat/p_compat_sys_request_key/p_compat_sys_request_key.o \
                  modules/exploit_detection/syscalls/__x32/p_x32_sys_keyctl/p_x32_sys_keyctl.o \
                  modules/exploit_detection/syscalls/override/p_override_creds/p_override_creds.o \
                  modules/exploit_detection/syscalls/override/p_revert_creds/p_revert_creds.o \
                  modules/exploit_detection/syscalls/override/overlayfs/p_ovl_create_or_link/p_ovl_create_or_link.o \
                  modules/exploit_detection/syscalls/pCFI/p_mark_inode_dirty/p_mark_inode_dirty.o \
                  modules/exploit_detection/syscalls/pCFI/p_schedule/p_schedule.o \
                  modules/exploit_detection/syscalls/pCFI/p___queue_work/p___queue_work.o \
                  modules/exploit_detection/syscalls/pCFI/p_lookup_fast/p_lookup_fast.o \
                  modules/exploit_detection/syscalls/p_ttwu_do_wakeup/p_ttwu_do_wakeup.o \
                  modules/exploit_detection/syscalls/p_capable/p_capable.o \
                  modules/exploit_detection/syscalls/p_scm_send/p_scm_send.o \
                  modules/exploit_detection/p_exploit_detection.o \
                  p_lkrg_main.o
Commit msg: LKRG in-tree @ 1f9809db4bed4e33fcc3db31
Ctrl+c to quit, any other key to continue

/media/debuilder/git-lkrg/lkrg ~
[linux-5.10.y b0dc91c6c] LKRG in-tree @ 1f9809db4bed4e33fcc3db31
 139 files changed, 19480 insertions(+)
 create mode 100644 security/lkrg/Kconfig
 create mode 100644 security/lkrg/Makefile
 create mode 100644 security/lkrg/modules/comm_channel/p_comm_channel.c
 create mode 100644 security/lkrg/modules/comm_channel/p_comm_channel.h
 create mode 100644 security/lkrg/modules/database/CPU.c
 create mode 100644 security/lkrg/modules/database/FTRACE/p_ftrace_enable_sysctl/p_ftrace_enable_sysctl.c
 create mode 100644 security/lkrg/modules/database/FTRACE/p_ftrace_enable_sysctl/p_ftrace_enable_sysctl.h
 create mode 100644 security/lkrg/modules/database/FTRACE/p_ftrace_modify_all_code/p_ftrace_modify_all_code.c
 create mode 100644 security/lkrg/modules/database/FTRACE/p_ftrace_modify_all_code/p_ftrace_modify_all_code.h
 create mode 100644 security/lkrg/modules/database/JUMP_LABEL/p_arch_jump_label_transform/p_arch_jump_label_transform.c
 create mode 100644 security/lkrg/modules/database/JUMP_LABEL/p_arch_jump_label_transform/p_arch_jump_label_transform.h
 create mode 100644 security/lkrg/modules/database/JUMP_LABEL/p_arch_jump_label_transform_apply/p_arch_jump_label_transform_apply.c
 create mode 100644 security/lkrg/modules/database/JUMP_LABEL/p_arch_jump_label_transform_apply/p_arch_jump_label_transform_apply.h
 create mode 100644 security/lkrg/modules/database/arch/arm/p_arm_metadata.c
 create mode 100644 security/lkrg/modules/database/arch/arm/p_arm_metadata.h
 create mode 100644 security/lkrg/modules/database/arch/arm64/p_arm64_metadata.c
 create mode 100644 security/lkrg/modules/database/arch/arm64/p_arm64_metadata.h
 create mode 100644 security/lkrg/modules/database/arch/p_arch_metadata.c
 create mode 100644 security/lkrg/modules/database/arch/p_arch_metadata.h
 create mode 100644 security/lkrg/modules/database/arch/x86/MSR.h
 create mode 100644 security/lkrg/modules/database/arch/x86/p_switch_idt/p_switch_idt.c
 create mode 100644 security/lkrg/modules/database/arch/x86/p_switch_idt/p_switch_idt.h
 create mode 100644 security/lkrg/modules/database/arch/x86/p_x86_metadata.c
 create mode 100644 security/lkrg/modules/database/arch/x86/p_x86_metadata.h
 create mode 100644 security/lkrg/modules/database/p_database.c
 create mode 100644 security/lkrg/modules/database/p_database.h
 create mode 100644 security/lkrg/modules/exploit_detection/arch/arm/p_ed_arm_arch.h
 create mode 100644 security/lkrg/modules/exploit_detection/arch/arm64/p_ed_arm64_arch.h
 create mode 100644 security/lkrg/modules/exploit_detection/arch/x86/p_ed_x86_arch.h
 create mode 100644 security/lkrg/modules/exploit_detection/p_exploit_detection.c
 create mode 100644 security/lkrg/modules/exploit_detection/p_exploit_detection.h
 create mode 100644 security/lkrg/modules/exploit_detection/p_rb_ed_trees/p_rb_ed_pids/p_rb_ed_pids_tree.c
 create mode 100644 security/lkrg/modules/exploit_detection/p_rb_ed_trees/p_rb_ed_pids/p_rb_ed_pids_tree.h
 create mode 100644 security/lkrg/modules/exploit_detection/syscalls/__x32/p_x32_sys_keyctl/p_x32_sys_keyctl.c
 create mode 100644 security/lkrg/modules/exploit_detection/syscalls/__x32/p_x32_sys_keyctl/p_x32_sys_keyctl.h
 create mode 100644 security/lkrg/modules/exploit_detection/syscalls/caps/p_cap_task_prctl/p_cap_task_prctl.c
 create mode 100644 security/lkrg/modules/exploit_detection/syscalls/caps/p_cap_task_prctl/p_cap_task_prctl.h
 create mode 100644 security/lkrg/modules/exploit_detection/syscalls/caps/p_sys_capset/p_sys_capset.c
 create mode 100644 security/lkrg/modules/exploit_detection/syscalls/caps/p_sys_capset/p_sys_capset.h
 create mode 100644 security/lkrg/modules/exploit_detection/syscalls/compat/p_compat_sys_add_key/p_compat_sys_add_key.c
 create mode 100644 security/lkrg/modules/exploit_detection/syscalls/compat/p_compat_sys_add_key/p_compat_sys_add_key.h
 create mode 100644 security/lkrg/modules/exploit_detection/syscalls/compat/p_compat_sys_capset/p_compat_sys_capset.c
 create mode 100644 security/lkrg/modules/exploit_detection/syscalls/compat/p_compat_sys_capset/p_compat_sys_capset.h
 create mode 100644 security/lkrg/modules/exploit_detection/syscalls/compat/p_compat_sys_keyctl/p_compat_sys_keyctl.c
 create mode 100644 security/lkrg/modules/exploit_detection/syscalls/compat/p_compat_sys_keyctl/p_compat_sys_keyctl.h
 create mode 100644 security/lkrg/modules/exploit_detection/syscalls/compat/p_compat_sys_request_key/p_compat_sys_request_key.c
 create mode 100644 security/lkrg/modules/exploit_detection/syscalls/compat/p_compat_sys_request_key/p_compat_sys_request_key.h
 create mode 100644 security/lkrg/modules/exploit_detection/syscalls/keyring/p_key_change_session_keyring/p_key_change_session_keyring.c
 create mode 100644 security/lkrg/modules/exploit_detection/syscalls/keyring/p_key_change_session_keyring/p_key_change_session_keyring.h
 create mode 100644 security/lkrg/modules/exploit_detection/syscalls/keyring/p_sys_add_key/p_sys_add_key.c
 create mode 100644 security/lkrg/modules/exploit_detection/syscalls/keyring/p_sys_add_key/p_sys_add_key.h
 create mode 100644 security/lkrg/modules/exploit_detection/syscalls/keyring/p_sys_keyctl/p_sys_keyctl.c
 create mode 100644 security/lkrg/modules/exploit_detection/syscalls/keyring/p_sys_keyctl/p_sys_keyctl.h
 create mode 100644 security/lkrg/modules/exploit_detection/syscalls/keyring/p_sys_request_key/p_sys_request_key.c
 create mode 100644 security/lkrg/modules/exploit_detection/syscalls/keyring/p_sys_request_key/p_sys_request_key.h
 create mode 100644 security/lkrg/modules/exploit_detection/syscalls/override/overlayfs/p_ovl_create_or_link/p_ovl_create_or_link.c
 create mode 100644 security/lkrg/modules/exploit_detection/syscalls/override/overlayfs/p_ovl_create_or_link/p_ovl_create_or_link.h
 create mode 100644 security/lkrg/modules/exploit_detection/syscalls/override/p_override_creds/p_override_creds.c
 create mode 100644 security/lkrg/modules/exploit_detection/syscalls/override/p_override_creds/p_override_creds.h
 create mode 100644 security/lkrg/modules/exploit_detection/syscalls/override/p_revert_creds/p_revert_creds.c
 create mode 100644 security/lkrg/modules/exploit_detection/syscalls/override/p_revert_creds/p_revert_creds.h
 create mode 100644 security/lkrg/modules/exploit_detection/syscalls/pCFI/p___queue_work/p___queue_work.c
 create mode 100644 security/lkrg/modules/exploit_detection/syscalls/pCFI/p___queue_work/p___queue_work.h
 create mode 100644 security/lkrg/modules/exploit_detection/syscalls/pCFI/p_lookup_fast/p_lookup_fast.c
 create mode 100644 security/lkrg/modules/exploit_detection/syscalls/pCFI/p_lookup_fast/p_lookup_fast.h
 create mode 100644 security/lkrg/modules/exploit_detection/syscalls/pCFI/p_mark_inode_dirty/p_mark_inode_dirty.c
 create mode 100644 security/lkrg/modules/exploit_detection/syscalls/pCFI/p_mark_inode_dirty/p_mark_inode_dirty.h
 create mode 100644 security/lkrg/modules/exploit_detection/syscalls/pCFI/p_schedule/p_schedule.c
 create mode 100644 security/lkrg/modules/exploit_detection/syscalls/pCFI/p_schedule/p_schedule.h
 create mode 100644 security/lkrg/modules/exploit_detection/syscalls/p_call_usermodehelper/p_call_usermodehelper.c
 create mode 100644 security/lkrg/modules/exploit_detection/syscalls/p_call_usermodehelper/p_call_usermodehelper.h
 create mode 100644 security/lkrg/modules/exploit_detection/syscalls/p_call_usermodehelper/p_usermode_kernel_dep.h
 create mode 100644 security/lkrg/modules/exploit_detection/syscalls/p_call_usermodehelper_exec/p_call_usermodehelper_exec.c
 create mode 100644 security/lkrg/modules/exploit_detection/syscalls/p_call_usermodehelper_exec/p_call_usermodehelper_exec.h
 create mode 100644 security/lkrg/modules/exploit_detection/syscalls/p_capable/p_capable.c
 create mode 100644 security/lkrg/modules/exploit_detection/syscalls/p_capable/p_capable.h
 create mode 100644 security/lkrg/modules/exploit_detection/syscalls/p_do_exit/p_do_exit.c
 create mode 100644 security/lkrg/modules/exploit_detection/syscalls/p_do_exit/p_do_exit.h
 create mode 100644 security/lkrg/modules/exploit_detection/syscalls/p_generic_permission/p_generic_permission.c
 create mode 100644 security/lkrg/modules/exploit_detection/syscalls/p_generic_permission/p_generic_permission.h
 create mode 100644 security/lkrg/modules/exploit_detection/syscalls/p_install.c
 create mode 100644 security/lkrg/modules/exploit_detection/syscalls/p_install.h
 create mode 100644 security/lkrg/modules/exploit_detection/syscalls/p_scm_send/p_scm_send.c
 create mode 100644 security/lkrg/modules/exploit_detection/syscalls/p_scm_send/p_scm_send.h
 create mode 100644 security/lkrg/modules/exploit_detection/syscalls/p_search_binary_handler/p_search_binary_handler.c
 create mode 100644 security/lkrg/modules/exploit_detection/syscalls/p_search_binary_handler/p_search_binary_handler.h
 create mode 100644 security/lkrg/modules/exploit_detection/syscalls/p_seccomp/p_seccomp.c
 create mode 100644 security/lkrg/modules/exploit_detection/syscalls/p_seccomp/p_seccomp.h
 create mode 100644 security/lkrg/modules/exploit_detection/syscalls/p_security_ptrace_access/p_security_ptrace_access.c
 create mode 100644 security/lkrg/modules/exploit_detection/syscalls/p_security_ptrace_access/p_security_ptrace_access.h
 create mode 100644 security/lkrg/modules/exploit_detection/syscalls/p_sel_write_enforce/p_sel_write_enforce.c
 create mode 100644 security/lkrg/modules/exploit_detection/syscalls/p_sel_write_enforce/p_sel_write_enforce.h
 create mode 100644 security/lkrg/modules/exploit_detection/syscalls/p_set_current_groups/p_set_current_groups.c
 create mode 100644 security/lkrg/modules/exploit_detection/syscalls/p_set_current_groups/p_set_current_groups.h
 create mode 100644 security/lkrg/modules/exploit_detection/syscalls/p_sys_setfsgid/p_sys_setfsgid.c
 create mode 100644 security/lkrg/modules/exploit_detection/syscalls/p_sys_setfsgid/p_sys_setfsgid.h
 create mode 100644 security/lkrg/modules/exploit_detection/syscalls/p_sys_setfsuid/p_sys_setfsuid.c
 create mode 100644 security/lkrg/modules/exploit_detection/syscalls/p_sys_setfsuid/p_sys_setfsuid.h
 create mode 100644 security/lkrg/modules/exploit_detection/syscalls/p_sys_setgid/p_sys_setgid.c
 create mode 100644 security/lkrg/modules/exploit_detection/syscalls/p_sys_setgid/p_sys_setgid.h
 create mode 100644 security/lkrg/modules/exploit_detection/syscalls/p_sys_setns/p_sys_setns.c
 create mode 100644 security/lkrg/modules/exploit_detection/syscalls/p_sys_setns/p_sys_setns.h
 create mode 100644 security/lkrg/modules/exploit_detection/syscalls/p_sys_setregid/p_sys_setregid.c
 create mode 100644 security/lkrg/modules/exploit_detection/syscalls/p_sys_setregid/p_sys_setregid.h
 create mode 100644 security/lkrg/modules/exploit_detection/syscalls/p_sys_setresgid/p_sys_setresgid.c
 create mode 100644 security/lkrg/modules/exploit_detection/syscalls/p_sys_setresgid/p_sys_setresgid.h
 create mode 100644 security/lkrg/modules/exploit_detection/syscalls/p_sys_setresuid/p_sys_setresuid.c
 create mode 100644 security/lkrg/modules/exploit_detection/syscalls/p_sys_setresuid/p_sys_setresuid.h
 create mode 100644 security/lkrg/modules/exploit_detection/syscalls/p_sys_setreuid/p_sys_setreuid.c
 create mode 100644 security/lkrg/modules/exploit_detection/syscalls/p_sys_setreuid/p_sys_setreuid.h
 create mode 100644 security/lkrg/modules/exploit_detection/syscalls/p_sys_setuid/p_sys_setuid.c
 create mode 100644 security/lkrg/modules/exploit_detection/syscalls/p_sys_setuid/p_sys_setuid.h
 create mode 100644 security/lkrg/modules/exploit_detection/syscalls/p_sys_unshare/p_sys_unshare.c
 create mode 100644 security/lkrg/modules/exploit_detection/syscalls/p_sys_unshare/p_sys_unshare.h
 create mode 100644 security/lkrg/modules/exploit_detection/syscalls/p_ttwu_do_wakeup/p_ttwu_do_wakeup.c
 create mode 100644 security/lkrg/modules/exploit_detection/syscalls/p_ttwu_do_wakeup/p_ttwu_do_wakeup.h
 create mode 100644 security/lkrg/modules/exploit_detection/syscalls/p_wake_up_new_task/p_wake_up_new_task.c
 create mode 100644 security/lkrg/modules/exploit_detection/syscalls/p_wake_up_new_task/p_wake_up_new_task.h
 create mode 100644 security/lkrg/modules/hashing/p_lkrg_fast_hash.c
 create mode 100644 security/lkrg/modules/hashing/p_lkrg_fast_hash.h
 create mode 100644 security/lkrg/modules/integrity_timer/p_integrity_timer.c
 create mode 100644 security/lkrg/modules/integrity_timer/p_integrity_timer.h
 create mode 100644 security/lkrg/modules/kmod/p_kmod.c
 create mode 100644 security/lkrg/modules/kmod/p_kmod.h
 create mode 100644 security/lkrg/modules/kmod/p_kmod_notifier.c
 create mode 100644 security/lkrg/modules/ksyms/p_resolve_ksym.c
 create mode 100644 security/lkrg/modules/ksyms/p_resolve_ksym.h
 create mode 100644 security/lkrg/modules/notifiers/p_notifiers.c
 create mode 100644 security/lkrg/modules/notifiers/p_notifiers.h
 create mode 100644 security/lkrg/modules/print_log/p_lkrg_debug_log.c
 create mode 100644 security/lkrg/modules/print_log/p_lkrg_log_level_shared.h
 create mode 100644 security/lkrg/modules/print_log/p_lkrg_print_log.h
 create mode 100644 security/lkrg/modules/self-defense/hiding/p_hiding.c
 create mode 100644 security/lkrg/modules/self-defense/hiding/p_hiding.h
 create mode 100644 security/lkrg/modules/wrap/p_struct_wrap.h
 create mode 100644 security/lkrg/p_lkrg_main.c
 create mode 100644 security/lkrg/p_lkrg_main.h
/media/debuilder/git-lkrg/lkrg/src /media/debuilder/git-lkrg/lkrg ~
```
Anyways, I have a question. 

I was trying to build the kernel using my regular config (the one before building lkrg as a regular module), and I see the following:

```
In file included from security/lkrg/modules/ksyms/p_resolve_ksym.c:19:
security/lkrg/modules/ksyms/../../p_lkrg_main.h:332:3: error: #error "LKRG requires CONFIG_KPROBES"
  332 |  #error "LKRG requires CONFIG_KPROBES"
      |   ^~~~~
security/lkrg/modules/ksyms/../../p_lkrg_main.h:338:3: error: #error "LKRG requires CONFIG_MODULE_UNLOAD"
  338 |  #error "LKRG requires CONFIG_MODULE_UNLOAD"
      |   ^~~~~
security/lkrg/modules/ksyms/../../p_lkrg_main.h:342:3: error: #error "LKRG requires CONFIG_KALLSYMS_ALL"
  342 |  #error "LKRG requires CONFIG_KALLSYMS_ALL"
      |   ^~~~~

```

But in the lkrg dependency is only `Depends on: SECURITY [=y]` -- wouldn't be better to add also the other deps to avoid such error while builing the kernel?

**Comment by solardiz** [comment]
> ./scripts/copy-builtin.sh: 54: pushd: not found

The script assumes bash. The first line should be changed to `#!/bin/bash`.

> ```
> In file included from security/lkrg/modules/ksyms/p_resolve_ksym.c:19:
> security/lkrg/modules/ksyms/../../p_lkrg_main.h:332:3: error: #error "LKRG requires CONFIG_KPROBES"
>   332 |  #error "LKRG requires CONFIG_KPROBES"
>       |   ^~~~~
> security/lkrg/modules/ksyms/../../p_lkrg_main.h:338:3: error: #error "LKRG requires CONFIG_MODULE_UNLOAD"
>   338 |  #error "LKRG requires CONFIG_MODULE_UNLOAD"
>       |   ^~~~~
> security/lkrg/modules/ksyms/../../p_lkrg_main.h:342:3: error: #error "LKRG requires CONFIG_KALLSYMS_ALL"
>   342 |  #error "LKRG requires CONFIG_KALLSYMS_ALL"
>       |   ^~~~~
> ```
> 
> But in the lkrg dependency is only `Depends on: SECURITY [=y]` -- wouldn't be better to add also the other deps to avoid such error while builing the kernel?

Yes, I think so.

@sempervictus If you agree, we'd appreciate a PR from you making these changes. Thanks!

**Comment by solardiz** [comment]
This issue is fixed in #44. Closing.
